OpenRA.Game/Sound/Sound.cs

```diff
@@ -348,9 +350,21 @@ public float VideoSeekPosition
 			var name = prefix + clip + suffix;
 
 			if (!string.IsNullOrEmpty(name) && (p == null || p == p.World.LocalPlayer))
-				soundEngine.Play2D(sounds[name],
+			{
+				ISound sound = soundEngine.Play2D(sounds[name],
 					false, relative, pos,
 					InternalSoundVolume * volumeModifier, attenuateVolume);
+				if (id != 0)
+				{
+					if (currentSounds.ContainsKey(id))
+					{
+						soundEngine.StopSound(currentSounds[id]);
+						currentSounds.Remove(id);
+					}
+
+					currentSounds.Add(id, sound);
```

Comment:
So far Remove() is working fine. Looks like it doesn't do anything if it the key doesn't exists.